### PR TITLE
chore: add a generic error for limit's exeeded

### PR DESCRIPTION
**Pull request by gastonfournier**
## About the changes
We don't have a meaningful error for limits established by the application. This could be a good starting point.

The error code is 400 cause I couldn't find anything better. 

The name of the error was picked from UnleashApiErrorTypes: https://github.com/Unleash/unleash/blob/2d8e9f87ff96fcb2e2f84a8b4b36bf5c0e28a885/src/lib/error/unleash-error.ts#L4-L34

### CODE REVIEW

File: src/lib/error/exeeds-limit-error.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { GenericUnleashError } from './unleash-error';
+
+export class ExeedsLimitError extends GenericUnleashError {
```
Comment (nunogois) [inline src/lib/error/exeeds-limit-error.ts:3]: Typo, should be `ExceedsLimitError`, both in the class name and the file.

File: src/lib/error/exeeds-limit-error.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    constructor(resource: string, limit: number) {
+        super({
+            name: 'ValidationError', // it can also be 'OperationDeniedError' or we can create a new one
```
Comment (nunogois) [inline src/lib/error/exeeds-limit-error.ts:6]: Why not call it `ExceedsLimitError`?

**Review by nunogois — APPROVED** [review]
I think this is a good starting point! Still, I think it's worth taking the comments into consideration before merging it.

File: src/lib/error/exeeds-limit-error.ts (hunk 2 above)
Comment (gastonfournier) [reply src/lib/error/exeeds-limit-error.ts]: I did, let's see how it goes :D

File: src/lib/error/exceeds-limit-error.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { GenericUnleashError } from './unleash-error';
+
+export class ExceedsLimitError extends GenericUnleashError {
+    constructor(resource: string, limit: number) {
+        super({
+            name: 'ExceedsLimitError', // it can also be 'OperationDeniedError' or we can create a new one
```
Comment (gastonfournier) [inline src/lib/error/exceeds-limit-error.ts:6]:
```suggestion
            name: 'ExceedsLimitError',
```

**Pull request merged**